### PR TITLE
add graphite-cli package

### PR DESCRIPTION
**Pull request by jrobsonchase**
###### Description of changes

Simple node2nix package, with some minor metadata fixups.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/graphite-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{ pkgs }:
```
Comment (AndersonTorres) [inline pkgs/development/tools/graphite-cli/default.nix:1]:
```suggestion
{ pkgs, lib }:
```

File: pkgs/development/tools/graphite-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+let
+  nodePackages = pkgs.callPackage ./node_packages { };
+  graphite-cli = nodePackages."@withgraphite/graphite-cli";
+in
+graphite-cli // {
+  meta = graphite-cli.meta // {
+    name = "graphite-cli";
+    license = "AGPL";
```
Comment (AndersonTorres) [inline pkgs/development/tools/graphite-cli/default.nix:9]:
```suggestion
    license = lib.licenses.agpl3Plus;
```

File: pkgs/development/tools/graphite-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    description = "Graphite is a fast, simple code review platform designed for engineers who want to write and review smaller pull requests, stay unblocked, and ship faster.";
```
Comment (AndersonTorres) [inline pkgs/development/tools/graphite-cli/default.nix:10]:
```suggestion
    description = "Fast, simple code review platform designed for engineers who want to write and review smaller pull requests, stay unblocked, and ship faster";
```

According to Contributing.md

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Now squash these commits into one.

**Comment by AndersonTorres** [comment]
Again, use `git rebase` to squash the commits into only one.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Again, use `git rebase` to squash the commits into only one.

**Comment by ryantm** [comment]
It appears that this is already in nixpkgs under `nodePackages.graphite-cli`. Since this is a user-facing CLI tool, it would be nice to add it to the top level so hydra builds it though. So maybe just have an entry like `graphite-cli = nodePackages.graphite-cli` in all-packages?

**Comment by jrobsonchase** [comment]
Cool, thanks for the useful suggestion @ryantm!

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Reword the commit title:

`nodePackages.graphite-cli: bring to top-level`

Besides, LGTM

**Comment by ryantm** [comment]
Result of `nixpkgs-review pr 211604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>graphite-cli</li>
  </ul>
</details>

**Pull request merged**